src/specFunctions.py

```diff
@@ -30,413 +30,426 @@
+    except:
+        errors = 1
+
+    try:
+        if (len(pixels) != len(wavelengths)):
+            #The Calibration points are of unequal length!
+            errors = 1
+        if (len(pixels) < 3):
+            #The Cal data contains less than 3 pixels!
+            errors = 1
+        if (len(wavelengths) < 3):
+            #The Cal data contains less than 3 wavelengths!
+            errors = 1
+    except:
+        errors = 1
+
+    if errors == 1:
+        print("Loading of Calibration data failed (missing caldata.txt or corrupted data!")
+        print("Loading placeholder data...")
+        print("You MUST perform a Calibration to use this software!\n\n")
+        pixels = [0,width/2,width]
+        wavelengths = [380,560,750]
+
+    wavelengthData, _, _ = compute_wavelengths(width, pixels, wavelengths, errors)
+
+    if message == 0:
+        calmsg1 = "UNCALIBRATED!"
+        calmsg2 = "Defaults loaded"
+        calmsg3 = "Perform Calibration!"
+    if message == 1:
+        calmsg1 = "Calibrated!!"
+        calmsg2 = "Using 3 cal points"
+        calmsg3 = "2nd Order Polyfit"
+    if message == 2:
+        calmsg1 = "Calibrated!!!"
+        calmsg2 = "Using > 3 cal points"
+        calmsg3 = "3rd Order Polyfit"
+
+    return [wavelengthData, calmsg1, calmsg2, calmsg3]
+
+
+def closest(lst, K):
+    # https://www.geeksforgeeks.org/python-find-closest-number-to-k-in-given-list/
+    return lst[min(range(len(lst)), key = lambda i: abs(lst[i]-K))]
+
+
+
+def writecal(clickArray, frameWidth):
+    calcomplete = False
+    pxdata = []
+    wldata = []
+    print("Enter known wavelengths for observed pixels!")
+    for _, _, pixel, wavelength_displayed in sorted(clickArray, key=lambda x: x[0]):
+        wavelength = input("Enter wavelength for {}px ({:.1f}nm): ".format(pixel, wavelength_displayed))
+        pxdata.append(pixel)
+        wldata.append(wavelength)
+
+    #This try except serves two purposes
+    #first I want to write data to the caldata.txt file without quotes
+    #second it validates the data in as far as no strings were entered 
+    try:
+        wldata = [float(x) for x in wldata]
+    except:
+        print("Only ints or decimals are allowed!")
+        print("Calibration aborted!")
+
+    _, _, calibration_error = compute_wavelengths(frameWidth, pxdata, wldata, 0)
+    while True:
+        save = input("R-squared={}. Overwrite current calibration data and save new values? ".format(calibration_error))
+        if save == "n":
+            return False
+        elif save != "y":
+            print("Invalid value '{}' entered. Please enter 'y' or 'n'.".format(save))
+            continue
+        else:
+            break
+
+    pxdata = ','.join(map(str, pxdata)) #convert array to string
+    wldata = ','.join(map(str, wldata)) #convert array to string
+    with open('caldata.txt','w') as f:
+        f.write(str(frameWidth)+'\r\n')
```

Comment:
Include the current frameWidth with the data. This way if the frame size is later changed, the calibration data still applies.